conans/client/manager.py

```diff
@@ -91,27 +91,29 @@ def deps_install(app, ref_or_path, install_folder, graph_info, remotes=None, bui
         manifest_manager.print_log()
 
     if install_folder:
-        conanfile.install_folder = install_folder
+        conanfile.set_base_install_folder(install_folder)
```

Comment:
Maybe using the ``install_folder`` as the place for putting some generators files (build systems generators) is not the best approach, and those generators need to use the ``build_folder`` instead?

Reply:
Yes, might be a possibility, not sure if it would be confusing for the user.